assets/js/base/components/state-input/state-input.js

```diff
@@ -28,7 +28,14 @@ const StateInput = ( {
 				name: decodeEntities( countryStates[ key ] ),
 		  } ) )
 		: [];
-
+	// @todo: remove this code block when issue https://github.com/woocommerce/woocommerce/issues/25854 is merged
+	// Defaults to the first state when selecting a country with states, this is here
+	// until a bug in Woo core is fixed.
+	useEffect( () => {
```

Comment:
We could add a link to the GitHub issue, so if we ever fix this in the code, we will remember to close the issue. :slightly_smiling_face: 
```suggestion
	// see: https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/1919
	useEffect( () => {
```